### DISCUSSION TITLE
No matching bindings found for serviceIdentifier: Array

### DISCUSSION THREAD

**Discussion by illia-halchun** [eclipse-glsp/glsp#851] [Q&A]
<img width="392" alt="Screenshot 2023-01-05 at 05 49 16" src="https://user-images.githubusercontent.com/93126099/210762774-be27fdb3-80ff-41a2-9b51-d8d11ab1484d.png">
<img width="1278" alt="Screenshot 2023-01-05 at 05 50 29" src="https://user-images.githubusercontent.com/93126099/210763029-ef818573-90ac-41d9-ad38-88e1be75deac.png">

**Comment by planger** [comment]
Hi Illia,

thank you for your message!
On a first glance, this doesn't seem to be an actual GLSP bug but rather an issue with the dependency injection setup in your project adopting of GLSP.

If it is a GLSP bug, can you please give us more concrete steps to reproduce this issue, e.g. based on the [GLSP project templates](https://github.com/eclipse-glsp/glsp-examples/tree/master/project-templates).

**Comment by illia-halchun** [comment]
<img width="243" alt="Screenshot 2023-01-05 at 14 03 47" src="https://user-images.githubusercontent.com/93126099/210860268-ce3280a1-f154-4dfc-a3d0-0ac1caf021d5.png">
Hi! I dont know whats wrong, because i just use standard layer architecture :) thank you in advance!

**Comment by planger** [comment]
It seems this issue is caused by code in your application and not really GLSP code. So unfortunately, without looking at your code (or at code based on the public GLSP examples that reproduces the issues), we can't really say what exactly is wrong.

Is the error only appearing if you add the `@inject(ENGINE_TYPES.EngineController) private EngineController: IEngineController` and is gone if you remove this constructor parameter injection?
If yes, you are probably missing to bind an implementation to that interface in your dependency injection module like this `bind(ENGINE_TYPES.EngineController).to(YOUR_IMPLEMENTATION_OF_THE_INTERFACE_IEngineController)`.

But as mentioned above, I can't be sure without seeing your code and without being able to reproduce this issue.

**Comment by illia-halchun** [comment]
<img width="500" alt="Screenshot 2023-01-05 at 14 37 12" src="https://user-images.githubusercontent.com/93126099/210865488-b324bf86-8ecf-4989-9b01-bfd415a0b76a.png">
https://github.com/IlliaHalchun/life-game-cli
Link the repo with source code, may it help :) Thank you for your help! I appreciate this!

**Reply by illia-halchun** [reply]
The issue occurs in /view/screen.view.ts on the 15th line

**Reply by planger** [reply]
I'm a bit confused, your repo doesn't even use or import GLSP, does it? Your issue seems to be related to plain inversify and unrelated to GLSP?
So it might be better to ask the question at https://github.com/inversify/InversifyJS

**Reply by illia-halchun** [reply]
> I'm a bit confused, your repo doesn't even use or import GLSP, does it? Your issue seems to be related to plain inversify and unrelated to GLSP? So it might be better to ask the question at https://github.com/inversify/InversifyJS

Oh, i'm so sorry about that, ive messed up reposytories...
Thanks for your attention ;)
